apps/test/unit/templates/curriculumCatalog/CurriculumCatalogTest.jsx

```diff
@@ -681,25 +684,46 @@ describe('CurriculumCatalog', () => {
           null
         )[0];
 
+        // Get the Stretch Recommended Curriculum for the current test curriculum. If the top stretch curriculum result is the same
+        // as the similar curriculum, then get the second stretch curriculum to match the logic in CurriculumCatalog.jsx.
+        const recommendedStretchCurriculumResults = getStretchRecommendations(
+          FULL_TEST_COURSES,
+          currCurriculum.key,
+          null
+        );
+        const recommendedStretchCurriculum =
```

Comment:
Is there a way we can make a test that forces the stretch curriculum to be the same as the recommended curriculum? It would be nice to test that that filtering works in the component itself.

Reply:
Sounds good! Is this what you were thinking? https://github.com/code-dot-org/code-dot-org/pull/57460/commits/51f79df0d8376ab76127f3c364f3e910c2b4b408

Reply:
not quite. I was more thinking of something that tested [this line](https://github.com/code-dot-org/code-dot-org/blob/f8df927d2751ed810efcd88bb426505dbff1fde2/apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx#L128). Unless I'm misunderstanding the test?

Reply:
I think the test I wrote checks that since it's figuring out what the Expected next result should be (either if the similar curriculum is the same as the stretch curriculum or not, and adjusts accordingly). `CurriculumCatalog` will then run the logic you linked to on [this line](https://github.com/code-dot-org/code-dot-org/blob/f8df927d2751ed810efcd88bb426505dbff1fde2/apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx#L128) and if the tests pass then the CurriculumCatalog is getting the curriculum we're expecting it to. I also check at the end to ensure that there is the right number of expected overlaps since when comparing against the 3rd curriculum, both Similar and Stretch return the same top curriculum. Please let me know if I'm misunderstanding this or what you're saying though!

Reply:
chatted offline and [these lines](https://github.com/code-dot-org/code-dot-org/blob/f8df927d2751ed810efcd88bb426505dbff1fde2/apps/test/unit/templates/curriculumCatalog/CurriculumCatalogTest.jsx#L713-L728) test that filtering!